packages/expo-media-library/src/MediaLibrary.ts

```diff
@@ -69,11 +69,17 @@ export type MediaLibraryAssetInfoQueryOptions = {
   shouldDownloadFromNetwork?: boolean;
 };
 
-export type MediaLibraryAssetChangeEvent = {
-  insertedAssets: Asset[];
-  deletedAssets: Asset[];
-  updatedAssets: Asset[];
-};
+export type MediaLibraryAssetChangeEvent =
+  /**
+   * iOS only
+   */
+  | {
+      assetPermissionsChanged: boolean;
+      insertedAssets: Asset[];
+      deletedAssets: Asset[];
+      updatedAssets: Asset[];
+    }
+  | object;
```

Comment:
I don't really get the idea of adding `| object` to this EventType? 🤔
I suggest going with the union type:
```
export type MediaLibraryAssetChangeEvent =
  | {
      assetPermissionsChanged: boolean;
    }
  | {
      insertedAssets: Asset[];
      deletedAssets: Asset[];
      updatedAssets: Asset[];
    }
```
And a side note: it should be `MediaLibraryAssetsChangeEvent` and not `MediaLibrary Asset ChangeEvent` as this event reports the change in multiple assets 🤔 
Do you mind creating a separate PR fixing this name? 😅